Add ADR for a query option to filter `get_urls`

philipnbbc opened the pull request:
# Details
This PR adds an ADR that proposes a `accept_get_urls` query option to filter flow segment `get_urls`. This allows clients to indicate that pre-signed URLs are not required, potentially significantly reducing the request time when requesting a large number of flow segments.

# Pivotal Story (if relevant)
Story URL: https://www.pivotaltracker.com/story/show/187930100

# Related PRs
_Where appropriate. Indicate order to be merged._

# Submitter PR Checks
_(tick as appropriate)_

- [ ] PR completes task/fixes bug
- [ ] API version has been incremented if necessary
- [ ] ADR status has been updated, and ADR implementation has been recorded
- [ ] Documentation updated (README, etc.)
- [ ] PR added to Pivotal story (if relevant)
- [ ] Follow-up stories added to Pivotal

# Reviewer PR Checks
_(tick as appropriate)_

- [ ] PR completes task/fixes bug
- [ ] Design makes sense, and fits with our current code base
- [ ] Code is easy to follow
- [ ] PR size is sensible
- [ ] Commit history is sensible and tidy

# Info on PRs
The checks above are guidelines. They don't all have to be ticked, but they should all have been considered.

AndrewGibb commented on docs/adr/0023-filter-segment-get-urls.md:
```diff
@@ -0,0 +1,78 @@
+
+## Context and Problem Statement
+
+Each Flow Segment returned by the `/flows/<flow id>/segment` endpoint includes a `get_urls` propery that may include (labelled) URLs for fetching media objects.
```
typo
```suggestion
Each Flow Segment returned by the `/flows/<flow id>/segment` endpoint includes a `get_urls` property that may include (labelled) URLs for fetching media objects.
```

AndrewGibb commented on api/TimeAddressableMediaStore.yaml:
```diff
@@ -1290,6 +1290,15 @@ paths:
+          description: |
+            A comma separated list of labels of flow segment `get_urls` to include in the response.
+            A 'default' label selects URLs with a 'default' or no label.
+            An empty `accept_get_urls` results in an empty or no `get_urls` flow segment property in the response.
```
Should we note, here in the description, the reason?
```suggestion
            An empty `accept_get_urls` results in an empty or no `get_urls` flow segment property in the response.
            Without `get_urls`, the response from the service is likely to be substantially faster.
```
?

philipnbbc replied:
I went with a slightly different wording in https://github.com/bbc/tams/pull/88/commits/d5ebe4a963071273b524439a55ceffc4241b94c3 because it depends on the number of flow segments and how optimised the pre-signed generation is.

AndrewGibb commented on api/TimeAddressableMediaStore.yaml:
```diff
@@ -1380,6 +1389,15 @@ paths:
+          description: |
+            A comma separated list of labels of flow segment `get_urls` to include in the response.
+            A 'default' label selects URLs with a 'default' or no label.
+            An empty `accept_get_urls` results in an empty or no `get_urls` flow segment property in the response.
```
Copy the change from L1298, if you accept it.

AndrewGibb reviewed (approved):
A suggested change and a typo. 

Once changed, LGTM.

philipnbbc commented:
This PR will be left open for now as further discussion is needed to understand whether it fits the requirements or not.

j616 commented:
Follow the Workshop where this was discussed, we should also update the [App Note](https://github.com/bbc/tams/blob/main/docs/appnotes/0009-storage-label-format.md) describing the `get_url` label format to add an extra `.presigned` element. i.e. `<provider>.<region[optional]>.<availabilityZone[optional]>:<storeType>.<presigned[optional]>:<storeName>` which may be used as such `example-cloud-provider.eu-west-1.a:example-storage-product.presigned:example-store-name`

philipnbbc commented:
> example-cloud-provider.eu-west-1.a:example-storage-product.presigned:example-store-name

Added in 

> Follow the Workshop where this was discussed, we should also update the [App Note](https://github.com/bbc/tams/blob/main/docs/appnotes/0009-storage-label-format.md) describing the `get_url` label format to add an extra `.presigned` element. i.e. `<provider>.<region[optional]>.<availabilityZone[optional]>:<storeType>.<presigned[optional]>:<storeName>` which may be used as such `example-cloud-provider.eu-west-1.a:example-storage-product.presigned:example-store-name`

Added in https://github.com/bbc/tams/pull/88/commits/ae1f3030ae69fb38e9aa77259a1b84319843e6f9

samdbmg commented on api/TimeAddressableMediaStore.yaml:
```diff
@@ -1513,13 +1513,13 @@ paths:
             Omitting `accept_get_urls` will result in all `get_urls` returned.
-            A 'default' label selects URLs with a 'default' or no label.
             An empty `accept_get_urls` results in an empty or no `get_urls` in the response.
+            Flow segment `get_urls` with no label cannot be filtered for.
```
Nit: Removes an ambiguity which could be interpreted as "always returned"
```suggestion
            Flow segment `get_urls` with no label cannot be filtered for; they will only be returned if `accept_get_urls` is omitted.
```

samdbmg commented on api/TimeAddressableMediaStore.yaml:
```diff
@@ -1615,13 +1615,13 @@ paths:
             Omitting `accept_get_urls` will result in all `get_urls` returned.
-            A 'default' label selects URLs with a 'default' or no label.
             An empty `accept_get_urls` results in an empty or no `get_urls` in the response.
+            Flow segment `get_urls` with no label cannot be filtered for.
```
Same nit as above
```suggestion
            Flow segment `get_urls` with no label cannot be filtered for; they will only be returned if `accept_get_urls` is omitted.
```
